src/lib/asl.c

```diff
@@ -0,0 +1,56 @@
+#include <asl.h>
+#include <pwd.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <time.h>
+
+#include <SystemConfiguration/SystemConfiguration.h>
+
+#include <xhyve/asl.h>
+
+static aslclient asl = NULL;
+static aslmsg log_msg = NULL;
+
+static unsigned char *buf = NULL;
+static size_t buf_size = 0;
```

Comment:
`buf_size` seems misnamed, as it is not the size of the buffer, but rather the index into the buffer where to write.

Reply:
That would depend on what you call size.  Size _is_ the size of the contents of the buffer (0 when it's empty).  Capacity is the "available size".